### ISSUE TITLE
filtering on number field

### ISSUE THREAD

**Issue by goravbhootra** [rcdilorenzo/filtrex#51]
Hi,

I am trying to create filters on a number field but keep encountering error with message `no with clause matching: {:error, "Invalid number value for '2'"}`

url: `http://localhost:5000/privileged/candidates?access_id=1ae51ba3-ee00-70b9-bc4a-57e99aaaf05b&page=3&status_equals=2`

model - candidate.ex
```
 defmodule MyApp.Members.Candidate do
   use Ecto.Schema
   use Arc.Ecto.Schema
   import Ecto.Changeset
   import Filtrex.Type.Config

...

  def filter_options(:privileged) do
    defconfig do
      number :status, allow_decimal: false
    end
  end
...
```

candidates_controller.ex
```
  ...
  {:ok, filter} <- Filtrex.parse_params(Candidate.filter_options(:privileged),
                                               Map.drop(params, ~w(access_id page))) do
  ...
    candidates =
      query
      |> Members.batch_candidates(batch.id)
      |> Members.preload_nominations_for_candidates()
      |> Filtrex.query(filter)
      |> Repo.paginate(Map.put(params, :page_size, 8)) 
  ...
```

What am I doing wrong?

**Comment by rcdilorenzo** [comment]
Do you have a stack trace that you can post here that includes the error `no with clause matching...`?

**Comment by goravbhootra** [comment]
@rcdilorenzo Thanks for the prompt response.

If I remove the handling of 
```
{:error, error} -> render_error(conn, error, privileged_candidate_path(conn, :index, access_id: access_id))
```
following error shows up:
```
[debug] Processing with MyApp.Privileged.CandidateController.index/2
  Parameters: %{"access_id" => "1ae51ba3-ee00-70b9-bc4a-57e99aaaf05b", "page" => "3", "status_equals" => "2"}
  Pipelines: [:browser, :privileged]
[debug] QUERY OK source="batches" db=6.8ms
...query...
[info] Sent 500 in 1518ms
[error] #PID<0.2676.0> running MyApp.Endpoint (cowboy_protocol) terminated
Server: localhost:5000 (http)
Request: GET /privileged/candidates?access_id=1ae51ba3-ee00-70b9-bc4a-57e99aaaf05b&page=3&status_equals=2
** (exit) an exception was raised:
    ** (WithClauseError) no with clause matching: {:error, "Invalid number value for '2'"}
        (pcms) lib/pcms_web/controllers/privileged/candidate_controller.ex:17: MyApp.Privileged.CandidateController.index/3
        (pcms) lib/pcms_web/controllers/privileged/candidate_controller.ex:1: MyApp.Privileged.CandidateController.action/2
        (pcms) lib/pcms_web/controllers/privileged/candidate_controller.ex:1: MyApp.Privileged.CandidateController.phoenix_controller_pipeline/2
        (pcms) lib/pcms_web/endpoint.ex:1: MyApp.Endpoint.instrument/4
        (phoenix) lib/phoenix/router.ex:278: Phoenix.Router.__call__/1
        (pcms) lib/plug/error_handler.ex:64: MyApp.Router.call/2
        (pcms) lib/pcms_web/endpoint.ex:1: MyApp.Endpoint.plug_builder_call/2
        (pcms) lib/plug/debugger.ex:102: MyApp.Endpoint."call (overridable 3)"/2
        (pcms) lib/pcms_web/endpoint.ex:1: MyApp.Endpoint.call/2
        (plug) lib/plug/adapters/cowboy/handler.ex:16: Plug.Adapters.Cowboy.Handler.upgrade/4
        (cowboy) /Users/gorav/Work/Solutionize/projects/pcms/deps/cowboy/src/cowboy_protocol.erl:442: :cowboy_protocol.execute/4
```

If I retain handling of the error in the code, error message flashes on the page
`Invalid number value for '2'`

Let me know if you need any further details.

**Comment by goravbhootra** [comment]
looks like this is re-surfacing https://github.com/rcdilorenzo/filtrex/pull/49

**Comment by goravbhootra** [comment]
@rcdilorenzo just realized that above fix is not part of the latest release. using master branch seems to have fixed the issue. exploring further.

would it be possible to release a new version with the fix so that somebody else trying the library doesnt land up in same situation. I wasted couple of hours trying to figure out if there was an issue with my setup and then moving on to explore other libraries.

**Comment by rcdilorenzo** [comment]
Absolutely. I apologize for the inconvenience. I'll try to do a release on Monday if I can. I didn't have that PR in the GitHub Project tab so I thought everything was up-to-date.

**Comment by goravbhootra** [comment]
@rcdilorenzo thanks for the wonderful library and prompt responses. may be couple of url examples would help new users. I will try to raise a PR once I get better understanding.

**Comment by rcdilorenzo** [comment]
I've published the new version 0.4.2 that includes this fix.
